src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -187,7 +188,10 @@ static SemanticTokensHelpers()
                 // Therefore, we set both optional parameters to false.
                 var spans = await ClassifierHelper.GetClassifiedSpansAsync(
                     document, textSpan, cancellationToken, removeAdditiveSpans: false, fillInClassifiedSpanGaps: false).ConfigureAwait(false);
-                classifiedSpans.AddRange(spans);
+
+                // The spans returned to us may include some empty spans, which we don't care about.
+                var nonEmptySpans = spans.Where(s => !s.TextSpan.IsEmpty);
```

Comment:
What is the use case for empty spans that are classified?

Reply:
I can't seem to reproduce it anymore, but there was a file where somehow an empty span was being produced as part of a classification. Razor breaks with empty spans (via a failed assert), so I thought it'd be good to just include this here as a precaution. Empty spans are also unnecessary to return in the LSP case.

Reply:
This actually seems to be a bug in `AdjustSpans` - for now I filed https://github.com/dotnet/roslyn/issues/58658 to track, but it shouldn't block this PR.